### PR TITLE
Add ROMFS support for Esp32

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
this PR aims to add ROMFS support for ESP32. (filesystem_utc not test romfs)
below point is add
1) add burning/download  romfs script

2) mount romfs during system booting up

3) modify defconfig to configure romfs partition.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [33deeff154f76a6c01ddafd78c02b243e0a0b052] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [33deeff154f76a6c01ddafd78c02b243e0a0b052] - Code Rule Check OK.

File: build/tools/esp32/esptool_py/romfs_flash_address.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	romfs_addr=`printf 0x"%08X" $romfs_addr`
+#	romfs_addr=`printf 0x"%08X" [$romfs_addr+$flash_base_addr]`
+
+	romfs_addr=`printf 0x"%08X" $(($romfs_addr + ${flash_base_addr}))`
+fi
+echo $romfs_addr 
+
+
+
+
```
Comment (drashti304) [inline build/tools/esp32/esptool_py/romfs_flash_address.sh]: Please remove extra lines

**Comment by sunghan-chang** [comment]
partion -> partition

**Comment by sunghan-chang** [comment]
last one is not modifying Kconfig, it's defconfig

**Comment by sunghan-chang** [comment]
mout -> mount

**Comment by sunghan-chang** [comment]
@ChinaTizenRT Could you check conflicts first?

File: os/board/esp32_devkit/src/esp32_boot.c [hunk 2]
```diff
@@ -115,6 +115,13 @@ void esp32_devKit_mount_partions(void)
 		}
 	}
 #endif
+
+#ifdef CONFIG_FS_ROMFS
+	ret = mount(CONFIG_ESP32_AUTOMOUNT_ROMFS_DEVNAME,CONFIG_ESP32_AUTOMOUNT_ROMFS_MOUNTPOINT,"romfs", 0, NULL);
```
Comment (drashti304) [inline os/board/esp32_devkit/src/esp32_boot.c]: space required after ,

Comment (SRCN-LiQin) [inline os/board/esp32_devkit/src/esp32_boot.c]: ret is defined under CONFIG_FLASH_PARITION , then it's rely on the dependency of these two CONFIG, be careful.

**Comment by seinfra** [comment]
Target : [4b2161d60e9513135e0c6ac0f9a381f2d021f9fd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4b2161d60e9513135e0c6ac0f9a381f2d021f9fd] - Code Rule Check OK.

File: os/board/esp32_devkit/Kconfig [hunk 3]
```diff
@@ -232,3 +232,18 @@ config ESP32_AUTOMOUNT_USERFS_MOUNTPOINT
     ---help---
         Specifies the mount point where user r/w file system
         will be mounted at.
+
+config ESP32_AUTOMOUNT_ROMFS_DEVNAME
+    string "Device name of the partition for rom read only file system"
+    default "/dev/mtdblock1"
```
Comment (SRCN-LiQin) [inline os/board/esp32_devkit/Kconfig]: is this device name should obey some rule from the partition arrange and also the source code logic ? if so, better add guide of naming in below help

File: build/tools/esp32/esptool_py/Makefile [hunk 4]
```diff
@@ -76,6 +78,14 @@ ALL:flash
 
 APP:app-flash
 
+ROMFS:romfs
```
Comment (SRCN-LiQin) [inline build/tools/esp32/esptool_py/Makefile]: is it necessary to limit this romfs burning under the CONFIG_FS_ROMFS?

File: build/tools/esp32/esptool_py/romfs_flash_address.sh (hunk 1 above)
Comment (ChinaTizenRT) [reply build/tools/esp32/esptool_py/romfs_flash_address.sh]: fixed

File: build/tools/esp32/esptool_py/Makefile (hunk 4 above)
Comment (ChinaTizenRT) [reply build/tools/esp32/esptool_py/Makefile]: if disable CONFIG_FS_ROMFS, this function will fail definitely.

File: os/board/esp32_devkit/Kconfig (hunk 3 above)
Comment (ChinaTizenRT) [reply os/board/esp32_devkit/Kconfig]: okay, improved

File: os/board/esp32_devkit/src/esp32_boot.c (hunk 2 above)
Comment (ChinaTizenRT) [reply os/board/esp32_devkit/src/esp32_boot.c]: okay, fixed.

Comment (ChinaTizenRT) [reply os/board/esp32_devkit/src/esp32_boot.c]: good idea, fixed.

**Comment by seinfra** [comment]
Target : [33deeff154f76a6c01ddafd78c02b243e0a0b052] -

**Comment by seinfra** [comment]
Target : [33deeff154f76a6c01ddafd78c02b243e0a0b052] -

**Comment by seinfra** [comment]
Target : [72492d6a212fa73fef74b0c6f6a3197a60f85126] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1f67cb3d3f9568eb7efba52b9838afc296653da6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1f67cb3d3f9568eb7efba52b9838afc296653da6] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@ChinaTizenRT  You should remove unnecessary commit, `Merge branch 'master' into esp32_romfs_support
1f67cb3`

File: os/board/esp32_devkit/src/esp32_boot.c [hunk 5]
```diff
@@ -102,8 +102,11 @@ extern int board_ledc_setup(void);
 
 void esp32_devKit_mount_partions(void)
 {
-#ifdef CONFIG_FLASH_PARTITION
+#if defined CONFIG_FLASH_PARTITION || defined CONFIG_FS_ROMFS
```
[the remark this answers is not in the document]
Comment (ChinaTizenRT) [reply os/board/esp32_devkit/src/esp32_boot.c]: fixed.
tks,notice it later.

File: build/tools/esp32/esptool_py/romfs_flash_address.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (ChinaTizenRT) [reply build/tools/esp32/esptool_py/romfs_flash_address.sh]: fixed.

**Comment by seinfra** [comment]
Target : [f1b1eeb936e41792258a32d6db478fa5617f21ba] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f1b1eeb936e41792258a32d6db478fa5617f21ba] - Code Rule Check OK.

**Comment by ChinaTizenRT** [comment]
You should remove unnecessary commit, Merge branch 'master' into esp32_romfs_support 1f67cb3
    -- "git rebase master" solved the confilcts.

**Comment by seinfra** [comment]
Target : [2f8c79623cd96f528f6d241338d7d1006381d26b] - Code Rule Check (C++) OK.

File: build/tools/esp32/esptool_py/Makefile [hunk 7]
```diff
@@ -76,6 +78,14 @@ ALL:flash
...
+
+romfs: $(ROMFS_IMG) $(ESPTOOLPY_SRC)
+ifneq ($(ROMFS_OFFSET),-1)
+	@echo "Flashing app to serial port $(ESPPORT), offset $(ROMFS_OFFSET)..."
```
Comment (ChinaTizenRT) [inline build/tools/esp32/esptool_py/Makefile]: app -> romfs partition

**Comment by seinfra** [comment]
Target : [2f8c79623cd96f528f6d241338d7d1006381d26b] - Code Rule Check OK.

File: build/tools/esp32/esptool_py/Makefile (hunk 7 above)
Comment (ChinaTizenRT) [reply build/tools/esp32/esptool_py/Makefile]: fixed,  "app" ->  "romfs image"

**Comment by seinfra** [comment]
Target : [6dd0fdf17a48d077a809a9612ea9457315798e30] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6dd0fdf17a48d077a809a9612ea9457315798e30] - Code Rule Check (C++) OK.

**Pull request merged**